Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/ReflectionAdapter.cpp

```diff
@@ -73,74 +93,203 @@ namespace AZ::DocumentPropertyEditor
+                }
+
+                return nullptr;
+            }
+
+            Dom::Value GetContainerNode(ReflectionAdapterReflectionImpl* impl, const AZ::Dom::Path& path)
+            {
+                Dom::Value containerRow;
+                const auto& findContainerProcedure = [&](const AZ::Dom::Path& nodePath, const ContainerEntry& containerEntry)
+                {
+                    if (containerRow.IsNull() && containerEntry.m_container && containerEntry.m_container->m_container == m_container)
+                    {
+                        containerRow = impl->m_adapter->GetContents()[nodePath];
+                        return false; // We've found our container row, so stop the visitor
+                    }
+                    return true;
+                };
+
+                // Find the row that contains the PropertyEditor for our actual container (if it exists)
+                auto visitorFlags =
+                    Dom::PrefixTreeTraversalFlags::ExcludeChildPaths | Dom::PrefixTreeTraversalFlags::TraverseMostToLeastSpecific;
+                impl->m_containers.VisitPath(path, findContainerProcedure, visitorFlags);
+
+                if (containerRow.IsNode())
+                {
+                    // Look within the Row for a PropertyEditor that has a SerializedPath field.
+                    // This will be the container's editor w/ attributes.
+                    for (auto it = containerRow.ArrayBegin(); it != containerRow.ArrayEnd(); ++it)
+                    {
+                        if (it->IsNode() && it->GetNodeName() == GetNodeName<Nodes::PropertyEditor>())
+                        {
+                            auto serializedFieldIt = it->FindMember(Reflection::DescriptorAttributes::SerializedPath);
+                            if (serializedFieldIt != it->MemberEnd())
+                            {
+                                return *it;
+                            }
+                        }
+                    }
+                }
+                return {};
             }
 
             void OnRemoveElement(ReflectionAdapterReflectionImpl* impl, const AZ::Dom::Path& path)
             {
-                AZ_Assert(m_elementInstance != nullptr, "Attempted to remove an element without a defined element instance");
-                m_container->RemoveElement(m_instance, m_elementInstance, impl->m_serializeContext);
+                void* instanceToRemove = m_elementInstance;
+
+                auto associativeContainer = m_container->GetAssociativeContainerInterface();
+                if (associativeContainer)
+                {
+                    const AZ::SerializeContext::ClassElement* containerClassElement =
+                        m_container->GetElement(m_container->GetDefaultElementNameCrc());
+                    instanceToRemove = associativeContainer->GetElementByKey(m_containerInstance, containerClassElement, m_elementInstance);
```

Comment:
The `m_elementInstance` in this case is actually what needs to be removed and not the key.
This should just be passed directly to `IDataContainer::RemoveElement`

Reply:
Removed the GetElementByKey block. I didn't have it originally, but thought there might be a case which needed it. You're right; we don't store the key in the LegacyReflectionBridge, only the element instance.